app/src/main/java/com/programmers/library/Mode.java

```diff
@@ -0,0 +1,41 @@
+package com.programmers.library;
+
+import com.programmers.library.controller.Controller;
+import com.programmers.library.repository.FileRepository;
+import com.programmers.library.repository.TestRepository;
+import com.programmers.library.service.Service;
+
+public enum Mode {
+    NORMAL(1, "일반"),
+    TEST(2, "테스트");
+
+    private final int id;
+    private final String name;
+    private final Controller controller;
+
+    Mode(int id, String name) {
+        this.id = id;
+        this.name = name;
+        this.controller = id == 2 ? new Controller(new Service(new TestRepository())) : new Controller(new Service(new FileRepository()));
+    }
```

Comment:
```suggestion
    NORMAL(1, "일반", new Controller(new Service(new FileRepository()))),
    TEST(2, "테스트", new Controller(new Service(new TestRepository())));

    private final int id;
    private final String name;
    private final Controller controller;

    Mode(int id, String name, Controller controller) {
        this.id = id;
        this.name = name;
        this.controller = controller;
    }
```

의도하신 방법이 이러한 방법이 아닌가 싶습니다. 타입의 다형성을 관리하기 위한 하나의 방법으로도 사용이 가능합니다.
현재는 이러한 방법으로 구현해놓고 고민해보시죠